cdragontoolbox/hashes.py

```diff
@@ -328,6 +341,35 @@ def substitute_basename_words(self):
     def add_basename_word(self):
         super()._add_basename_word(self.build_wordlist())
 
+    def multi_substitution(self, plugin, fileext=""):
+        """Builds a list of paths based on the specified plugin and file-extension.
+        Replaces a word in all basenames in these paths while adding an additional word near it."""
+
+        words = self.build_wordlist()
+        paths = set()
+        for path in self.known.values():
+            if path.startswith(f"plugins/{plugin}") and path.endswith(fileext):
```

Comment:
- You can factorize `f"plugins/{plugin}"`.
- `paths` can be a list (there will be no duplicates to remove)